lib/src/Uncertainty/Distribution/SmoothedUniform.cxx

```diff
@@ -0,0 +1,436 @@
+ *
+ */
+#include <cmath>
+#include "openturns/SmoothedUniform.hxx"
+#include "openturns/Uniform.hxx"
+#include "openturns/Normal.hxx"
+#include "openturns/DistFunc.hxx"
+#include "openturns/RandomGenerator.hxx"
+#include "openturns/PersistentObjectFactory.hxx"
+
+
+BEGIN_NAMESPACE_OPENTURNS
+
+CLASSNAMEINIT(SmoothedUniform)
+
+static const Factory<SmoothedUniform> Factory_SmoothedUniform;
+
+/* Default constructor */
+SmoothedUniform::SmoothedUniform()
+  : RandomMixture()
+  , a_(-1.0)
+  , b_(1.0)
+  , sigma_(1.0)
+{
+  setName( "SmoothedUniform" );
+  setDimension( 1 );
+  DistributionCollection coll(2);
+  coll[0] = Uniform(a_, b_);
+  coll[1] = Normal(0.0, sigma_);
+  setDistributionCollectionAndWeights(coll, Matrix(1, 2, Point(2, 1.0)), false);
+  computeRange();
+  setParallel(true);
+}
+
+/* Parameters constructor */
+SmoothedUniform::SmoothedUniform(const Scalar a,
+                 const Scalar b,
+                 const Scalar sigma)
+  : RandomMixture()
+  , a_(a)
+  , b_(b)
+  , sigma_(sigma)
+{
+  setName( "SmoothedUniform" );
+  setDimension( 1 );
+  if (b <= a) throw InvalidArgumentException(HERE) << "Error the upper bound b of the Uniform component of a SmoothedUniform distribution must be greater than its lower bound a, here a=" << a << " b=" << b;
+  if (sigma <= 0.0) throw InvalidArgumentException(HERE) << "Error the standard deviation of the Normal component of a SmoothedUniform must be positive, here sigma=" << sigma;
+  DistributionCollection coll(2);
+  coll[0] = Uniform(a_, b_);
+  coll[1] = Normal(0.0, sigma_);
+  setDistributionCollectionAndWeights(coll, Matrix(1, 2, Point(2, 1.0)), false);
+  computeRange();
+  setParallel(true);
+}
+
+/* Comparison operator */
+Bool SmoothedUniform::operator ==(const SmoothedUniform & other) const
+{
+  if (this == &other) return true;
+  return (a_ == other.a_) && (b_ == other.b_) && (sigma_ == other.sigma_);
+}
+
+Bool SmoothedUniform::equals(const DistributionImplementation & other) const
+{
+  const SmoothedUniform* p_other = dynamic_cast<const SmoothedUniform*>(&other);
+  return p_other && (*this == *p_other);
+}
+
+/* String converter */
+String SmoothedUniform::__repr__() const
+{
+  OSS oss(true);
+  oss << "class=" << SmoothedUniform::GetClassName()
+      << " name=" << getName()
+      << " dimension=" << getDimension()
+      << " a=" << a_
+      << " b=" << b_
+      << " sigma=" << sigma_;
+  return oss;
+}
+
+String SmoothedUniform::__str__(const String & offset) const
+{
+  OSS oss(false);
+  oss << offset << getClassName() << "(a = " << a_ << ", b = " << b_ << ", sigma = " << sigma_ << ")";
+  return oss;
+}
+
+/* Virtual constructor */
+SmoothedUniform * SmoothedUniform::clone() const
+{
+  return new SmoothedUniform(*this);
+}
+
+/* Get one realization of the distribution */
+Point SmoothedUniform::getRealization() const
+{
+  return Point(1, a_ + (b_ - a_) * RandomGenerator::Generate() + sigma_ * DistFunc::rNormal());
+}
```

Comment:
Seems that the translation factor a_ should be removed ? compare with getSample/computeMean